core/arch/arm/plat-imx6/imx6_init.S

```diff
@@ -0,0 +1,346 @@
+ */
+FUNC plat_cpu_reset_early , :
+UNWIND(	.fnstart)
+
+	/*
+	 * Mandated HW config loaded
+	 *
+	 * SCTLR = 0x00004000
+	 * - Round-Robin replac. for icache, btac, i/duTLB (bit14: RoundRobin)
+	 *
+	 * ACTRL = 0x00000041
+	 * - core always in full SMP (FW bit0=1, SMP bit6=1)
+	 * - L2 write full line of zero disabled (bit3=0)
+	 *   (keep WFLZ low. Will be set once outer L2 is ready)
+	 *
+	 * NSACR = 0x00020C00
+	 * - NSec cannot change ACTRL.SMP (NS_SMP bit18=0)
+	 * - Nsec can lockdown TLB (TL bit17=1)
+	 * - NSec cannot access PLE (PLE bit16=0)
+	 * - NSec can use SIMD/VFP (CP10/CP11) (bit15:14=2b00, bit11:10=2b11)
+	 *
+	 * PCR = 0x00000001
+	 * - no change latency, enable clk gating
+	 */
+	movw r0, #0x4000
+	movt r0, #0x0000
+	write_sctlr r0
+
+	movw r0, #0x0041
+	movt r0, #0x0000
+	write_actlr r0
+
+	movw r0, #0x0C00
+	movt r0, #0x0002
+	write_nsacr r0
+
+	movw r0, #0x0000
+	movt r0, #0x0001
+	write_pcr r0
+
+	/*
+	 * GIC configuration
+	 *
+	 * Register ICDISR0 = 0xFFFFFFFF
+	 * - All local interrupts are NonSecure.
+	 *
+	 * Register ICCPMR = 0xFFFFFFFF
+	 */
+
+	ldr  r0, =GIC_DIST_BASE
+	mov  r1, #0xFFFFFFFF
+	str  r1, [r0, #GIC_DIST_ISR0]
+
+	ldr  r0, =GIC_CPU_BASE
+	mov  r1, #0xFFFFFFFF
+	str  r1, [r0, #CORE_ICC_ICCPMR]
+
+	mov pc, lr /* back to tzinit */
+UNWIND(	.fnend)
+END_FUNC plat_cpu_reset_early
+
+/*
+ * A9 secured config, needed only from a single core
+ *
+ * Use scratables registers R0-R3.
+ * No stack usage.
+ * LR store return address.
+ * Trap CPU in case of error.
+ *
+ * TODO: size optim in code
+ */
+FUNC plat_cpu_reset_late , :
+UNWIND(	.fnstart)
+
+	mrc p15, 0, r0, c0, c0, 5
+	ands r0, #3
+	beq _boot_late_primary_cpu
+
+_boot_late_secondary_cpu:
+	mov pc, lr
+
+_boot_late_primary_cpu:
+
+	/*
+	 * Snoop Control Unit configuration
+	 *
+	 * SCU is enabled with filtering off.
+	 * Both Secure/Unsecure can access SCU and timers
+	 *
+	 * 0x00 SCUControl = 0x00000060 !!!  should be 0x5 !
+	 * 0x04 SCUConfiguration =  ???
+	 * 0x0C SCUInvalidateAll (Secure cfg)
+	 * 0x40 FilteringStartAddress = 0x40000000
+	 * 0x44 FilteeringEndAddress - 0x80000000
+	 * 0x50 SCUAccessControl
+	 * 0x54 SCUSecureAccessControl
+	 */
+
+	/*
+	 * SCU Access Register : SAC = 0x0000000F
+	 * - both secure CPU access SCU
+	 */
+	ldr  r0, =SCU_BASE
+	movw r1, #0x000F
+	movt r1, #0x0000
+	str  r1, [r0, #SCU_SAC]
+
+	/*
+	 * SCU NonSecure Access Register : SNSAC : 0x00000FFF
+	 * - both nonsec cpu access SCU, private and global timer
+	 */
+	movw r1, #0x0FFF
+	movt r1, #0x0000
+	str  r1, [r0, #SCU_NSAC]
+
+	/*- GIC secure configuration ---*/
+
+	/*
+	 * Register ICDISR[1-31] = 0xFFFFFFFF
+	 * - All external interrupts are NonSecure.
+	 */
+	ldr  r0, =(GIC_DIST_BASE + GIC_DIST_ISR1)
+	mov  r2, #0xFFFFFFFF
+	mov  r1, #31 /* Nb of loop rounds */
+loop_1:
+	str r2, [r0]
+	add r0, #4
+	sub r1, r1, #1
+	cmp r1, #0
+	bne loop_1
+
+	/* Allow NSec to manage FIQ/Imprecise abort */
+	mrc p15, 0, r0, c1, c1, 0    /* read Secure Configuration Register */
```

Comment:
Actually I had doubt on this too; I copied this from the tz_a9init.S of plat-stm. The ARM doc says the combination of FW=1 and FIQ=1 should be avoided as apparently the NS world can mask out the FIQ thus denies FIQ to reach to OP-TEE.
However, it seems the OP-TEE does not use FIQ for this platform, and VxWorks also ignores FIQ (in GIC configuration all INTs are assigned to GROUP0), so letting NS to manage FIQs will not cause any problem. And in order to keep consistency I did change this code.